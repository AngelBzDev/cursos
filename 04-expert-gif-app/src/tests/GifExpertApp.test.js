import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp'

describe('Pruebas a GifExpertApp', () => {
   test('Debe mostrarse correctamente', () => {
      const wrapper = shallow(<GifExpertApp />);
      expect(wrapper).toMatchSnapshot();
   })
   test('Debe mostrar la lista de categorias', () => {
      
      const categories = ['Pokeon', 'Superman', 'Spider man'];
      const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
      
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find('GifGrid').length).toBe(categories.length);
   })
   
})

