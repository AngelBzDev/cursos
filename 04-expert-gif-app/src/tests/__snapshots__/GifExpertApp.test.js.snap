// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifExpertApp Debe mostrar la lista de categorias 1`] = `
<Fragment>
  <h2
    className="titulo"
  >
    Gif App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ul>
    <GifGrid
      category="Pokeon"
      key="Pokeon"
    />
    <GifGrid
      category="Superman"
      key="Superman"
    />
    <GifGrid
      category="Spider man"
      key="Spider man"
    />
  </ul>
</Fragment>
`;

exports[`Pruebas a GifExpertApp Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2
    className="titulo"
  >
    Gif App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ul />
</Fragment>
`;
